modules/guides/pages/import.adoc

```diff
@@ -0,0 +1,335 @@
+
+We now need to parse the input data, either by reading the whole document into memory (if that's possible) or streaming each record in turn.
+
+[{tabs}]
+====
+////
+pass:[.NET]::
++
+--
+[source,csharp]
+----
+include::dotnet-sdk:howtos:example$Import.cs[tag=connect,indent=0]
+----
+--
+Java::
++
+--
+[source,java]
+----
+include::java-sdk:howtos:example$Import.java[tag=connect,indent=0]
+----
+--
+
+////
+
+Node.js::
++
+--
+
+We can easily stream a *CSV* file with an external library.
+
+[source,nodejs]
+----
+include::nodejs-sdk:howtos:example$import.js[tags=csv-tsv-import,indent=0]
+----
+
+[source,nodejs]
+----
+include::nodejs-sdk:howtos:example$import.js[tags=csvStream,indent=0]
+----
+
+*TSV* files can be processed with the same library:
+
+[source,nodejs]
+----
+include::nodejs-sdk:howtos:example$import.js[tags=csv-tsv-import,indent=0]
+----
+
+[source,nodejs]
+----
+include::nodejs-sdk:howtos:example$import.js[tags=tsvStream,indent=0]
+----
+
+*JSON* requires a different library, which streams its output in a `{ key: ..., value: ...}` wrapper, so we need to map the stream into the expected format:
+
+[source,nodejs]
+----
+include::nodejs-sdk:howtos:example$import.js[tags=json-jsonl-import;json-import,indent=0]
+----
+----
+include::nodejs-sdk:howtos:example$import.js[tags=jsonStream,indent=0]
+----
+
+The *JSON Lines* format requires the same treatment:
+
+[source,nodejs]
+----
+include::nodejs-sdk:howtos:example$import.js[tags=json-jsonl-import;jsonl-import,indent=0]
+----
+----
+include::nodejs-sdk:howtos:example$import.js[tags=jsonlStream,indent=0]
+----
+--
+Python::
++
+--
+We can easily parse a *CSV* file with an external library.
+
+[source,python]
+----
+include::python-sdk:howtos:example$import.py[tags=csv-tsv-import;csvImport,indent=0]
+----
+
+*TSV* files can be processed with the same library:
+
+[source,python]
+----
+include::python-sdk:howtos:example$import.py[tags=csv-tsv-import;tsvImport,indent=0]
+----
+
+
+*JSON* requires a different library:
+
+[source,python]
+----
+include::python-sdk:howtos:example$import.py[tags=json-jsonl-import;jsonImport,indent=0]
+----
+
+The *JSON Lines* format requires the same treatment, with a little processing to get the line-by-line behaviour:
+
+[source,python]
+----
+include::python-sdk:howtos:example$import.py[tags=json-jsonl-import;jsonlImport,indent=0]
+----
+====
+
+
+
+== Inserting the documents
+
+So we have a document, and we know where to put it, let's go!
+But of course, Couchbase is a "Key-Value" store, and the document is the *value*... but what's the key?
+
+So the first step is to determine the key. This could be as simple as extracting the `id` field from the document, or using an incrementing sequence number.
+
+Then we may want to do additional processing, for example calculating fields, or adding metadata about the importer.
+
+Finally we do an insert (or upsert, to update the document with that key, if it exists.)
```

Comment:
Again, rewrite to be more of a how-to guide than a tutorial. Should be less hand-holdy, and definitely second person ("you do this!" not "we can do this".)

```suggestion
Having processed each imported document, you can insert it into the keyspace.
Couchbase is a key-value store, and the document is the value, so before you can insert the document, you need to determine the key.

To insert an imported document into the keyspace:

. Specify the key. This could be as simple as extracting the `id` field from the document, or using an incrementing sequence number.

. Do any additional processing, for example calculating fields, or adding metadata about the importer.

. Finally, use an insert operation to the insert the document.
Alternatively, if you want to update an existing document with the same key, use an upsert operation to update the existing document.
```